static/app/components/modals/savedSearchModal/createSavedSearchModal.tsx

```diff
@@ -38,7 +38,9 @@ function validateSortOption({
   );
   const sortOptions = [...DEFAULT_SORT_OPTIONS];
   if (hasBetterPrioritySort) {
-    sortOptions.push(IssueSortOptions.BETTER_PRIORITY);
+    sortOptions.unshift(IssueSortOptions.BETTER_PRIORITY);
+    const index = sortOptions.indexOf(IssueSortOptions.PRIORITY);
+    sortOptions.splice(index, 1);
   }
```

Comment:
can we declare the sort options inline and add either BETTER_PRIORITY or PRIORITY instead of trying to do a bunch of mutating